### PR TITLE
doc: Rework guix docs after 1.4 release

### LINKED ISSUE #28957: doc: update Guix install docs for 1.4.0
[Guix 1.4.0 has been out for nearly 1 year.](https://guix.gnu.org/en/blog/2022/gnu-guix-1.4.0-released/) Our current docs ([contrib/guix/INSTALL.md](https://github.com/bitcoin/bitcoin/blob/master/contrib/guix/INSTALL.md)) mostly describe installing 1.3.0, and it's related dependencies. Updating the guide for 1.4.0 could simplify many things. i.e rather than discussing Guile 3.0 vs 2.0, and installation instructions/build complication for either version, we [can settle on just 3.0](https://guix.gnu.org/manual/en/html_node/Requirements.html). I have done a from-source build of Guix, on Ubuntu Jammy, using distro packaged deps, and everything seems to be working as expected.

### PR DESCRIPTION
**Pull request by maflcko**
Follow-up to https://github.com/bitcoin/bitcoin/pull/28902

Fixes https://github.com/bitcoin/bitcoin/issues/28957

### CODE REVIEW

File: contrib/guix/INSTALL.md [hunk 1]
```diff
@@ -167,80 +154,41 @@ For reference, the graphic below outlines Guix v1.3.0's dependency graph:
 
 ![bootstrap map](https://user-images.githubusercontent.com/6399679/125064185-a9a59880-e0b0-11eb-82c1-9b8e5dc9950d.png)
 
-#### Consider /tmp on tmpfs
-
-If you use an NVME (SSD) drive, you may encounter [cryptic build errors](#coreutils-fail-teststail-2inotify-dir-recreate). Mounting a [tmpfs at /tmp](https://ubuntu.com/blog/data-driven-analysis-tmp-on-tmpfs) should prevent this and may improve performance as a bonus.
```
[the remark this answers is not in the document]
Comment (Sjors) [reply contrib/guix/INSTALL.md:172]: It took a _lot_ of debugging cryptic errors with the help of @dongcarl to figure out that the error I was running into was to due to trying to build guix on a physical SSD drive. It's also just faster to use a RAM drive.

**Comment by fanquake** [comment]
Concept ACK - seems fine to drop most of the workarounds / guile 2 vs 3 docs, as anyone building from scratch at this point, should be building 1.4.0 or master.

**Review by sedited — APPROVED** [review]
ACK fad444f6e1b1ce5c1572c773db4a9a098c7a0b96

**Review by fanquake — APPROVED** [review]
ACK fad444f6e1b1ce5c1572c773db4a9a098c7a0b96

**Pull request merged**